scripts/utils/options.ts

```diff
@@ -0,0 +1,245 @@
+/**
+ * Use commander and prompts to gather a list of options for a script
+ */
+
+import prompts from 'prompts';
+import type { PromptObject } from 'prompts';
+import program from 'commander';
+import type { Command } from 'commander';
+import kebabCase from 'lodash/kebabCase';
+
+// Option types
+
+export type OptionId = string;
+export type BaseOption = {
+  description?: string;
+  /**
+   * By default the one-char version of the option key will be used as short flag. Override here,
+   *   e.g. `shortFlag: 'c'`
+   */
+  shortFlag?: string;
+};
+
+export type BooleanOption = BaseOption & {
+  /**
+   * Does this option default true?
+   */
+  inverse?: boolean;
+};
+
+export type StringOption = BaseOption & {
+  values: string[];
+  required?: boolean;
+};
+
+export type StringArrayOption = BaseOption & {
+  values: string[];
+  multiple: true;
+};
+
+// StringArrayOption requires `multiple: true;` but unless you use `as const` an object with
+// { multiple: true } will be inferred as `multiple: boolean;`
+type StringArrayOptionMatch = Omit<StringArrayOption, 'multiple'> & { multiple: boolean };
+
+export type Option = BooleanOption | StringOption | StringArrayOption;
+export type MaybeOptionValue<TOption extends Option> = TOption extends StringArrayOptionMatch
+  ? string[]
+  : TOption extends StringOption
+  ? string | undefined
+  : TOption extends BooleanOption
+  ? boolean
+  : never;
```

Comment:
is it `never` or `unknown`?

Reply:
I don't know the difference 🙈 -- I would guess `never` because it is impossible?

Reply:
Well if the options spec object is properly validated then I guess it's impossible, otherwise I'd go for `unknown` because otherwise the type would be hidden from intellisense

Reply:
I mean it's not possible to pass an object into anything that returns an `OptionValue` that's not a valid `Option`, so I don't think it matters?